src/components/HelpText/helpers/positionResolver.js

```diff
@@ -1,60 +1,22 @@
-const bottomLeft = opts => {
-    const { trigger, content, viewport } = opts;
-    if (
-        trigger.leftBottomAnchor.x + content.width <= viewport.width &&
-        trigger.leftBottomAnchor.y + content.height <= viewport.height
-    ) {
-        return {
-            top: trigger.leftBottomAnchor.y,
-            left: trigger.leftBottomAnchor.x,
-        };
-    }
-    return false;
-};
-
-const bottomRight = opts => {
-    const { trigger, content, viewport } = opts;
-    if (
-        trigger.rightBottomAnchor.x - content.width >= 0 &&
-        trigger.leftBottomAnchor.y + content.height <= viewport.height
-    ) {
-        return {
-            top: trigger.leftBottomAnchor.y,
-            left: trigger.rightBottomAnchor.x - content.width,
-        };
-    }
-    return false;
-};
-
-const upLeft = opts => {
-    const { trigger, content, viewport } = opts;
-    if (
-        trigger.leftUpAnchor.x + content.width <= viewport.width &&
-        trigger.leftUpAnchor.y - content.height >= 0
-    ) {
-        return {
-            top: trigger.leftUpAnchor.y - content.height,
-            left: trigger.leftBottomAnchor.x,
-        };
-    }
-    return false;
-};
-
-const upRight = opts => {
-    const { trigger, content } = opts;
-    if (
-        trigger.rightUpAnchor.x - content.width >= 0 &&
-        trigger.rightUpAnchor.y - content.height >= 0
-    ) {
-        return {
-            top: trigger.leftUpAnchor.y - content.height,
-            left: trigger.rightUpAnchor.x - content.width,
-        };
-    }
-    return false;
-};
+import resolverBottomLeft from '../../InternalOverlay/helpers/resolverBottomLeft';
+import resolverBottomRight from '../../InternalOverlay/helpers/resolverBottomRight';
+import resolverUpLeft from '../../InternalOverlay/helpers/resolverUpLeft';
+import resolverUpRight from '../../InternalOverlay/helpers/resolverUpRight';
+import resolverBottomCenter from '../../InternalOverlay/helpers/resolverBottomCenter';
+import resolverUpCenter from '../../InternalOverlay/helpers/resolverUpCenter';
+import resolverCenterLeft from '../../InternalOverlay/helpers/resolverCenterLeft';
+import resolverCenterRight from '../../InternalOverlay/helpers/resolverCenterRight';
 
-const resolvers = [bottomRight, bottomLeft, upRight, upLeft];
+const resolvers = [
+    resolverBottomRight,
+    resolverBottomLeft,
+    resolverBottomCenter,
+    resolverUpRight,
+    resolverUpLeft,
+    resolverUpCenter,
+    resolverCenterRight,
+    resolverCenterLeft,
+];
```

Comment:
The internal overlay default resolver should use this by default, there is no need to repeat this here.
What we want is that in the great majority use cases of the InternalOverlay the consumer does not need to use the position resolver, but only for exceptional cases.

Reply:
In the component design it has 2 differences: first the margin is 0 and second the order of is different.

Reply:
@yvmunayev Design is only a representation and this is meant to not be used here in this component